### PR TITLE
fix in to_interchange_format

### PR DESCRIPTION
**Pull request by JGuetschow**
# Pull request

Please confirm that this pull request has done the following:

- [ ] Tests added
- [x] Documentation added (where applicable)
- [x] Description in a ``.rst`` file in the directory ``changelog_unreleased`` added - remember to include a ``*`` to make it a bullet point
No new functionality added, tests  are not affected. One could add a tests for the bug that has been fixed.

## Description

Fix in to_interchnage_format. Before timeseries which contained only nan were kept (I assume the intention was to remove them, as there was a dropna in the code). On the other hand time series which contained only a few (or single) data points were in some cases dropped. This has now been fixed by specifying the time columns as subset in dropna.

A further tiny fix is to make regexp_unit optional in nir_add_unit_information as was intended already before, but not indicated in the function header.

### CODE REVIEW

File: changelog_unreleased/fix-data-reading.rst [hunk 1]
```diff
@@ -0,0 +1 @@
+* Fix handling of units when reading data where CO2eq and native units coexist
```
Comment (mikapfl) [inline changelog_unreleased/fix-data-reading.rst:1]: ? I don't see any changes in the code related to this.

File: primap2/_merge.py [hunk 2]
```diff
@@ -54,8 +54,10 @@ def merge_with_tolerance_core(
     da_merge: xr.DataArray
         data to merge into ds_start
     tolerance: float (optional), default = 0.01
-        The tolerance to use when comparing data
-    error_on_discrepancy: (optional), default = True
+        The tolerance to use when comparing data. Tolerance is relative to values in
+        the calling Dataset. Thus by default a 1% deviation of values in da_merge
+        from the calling Dataset is tolerated.
+    error_on_discrepancy (optional), default = True
```
Comment (mikapfl) [inline primap2/_merge.py:60]:
```suggestion
    error_on_discrepancy: (optional), default = True
```
Why remove the `:`?

File: primap2/_merge.py [hunk 3]
```diff
@@ -253,8 +255,10 @@ def merge(
         da_merge: xr.DataArray
             data to merge to the calling object
         tolerance: float (optional), default = 0.01
-            The tolerance to use when comparing data
-        error_on_discrepancy: (optional), default = True
+            The tolerance to use when comparing data. Tolerance is relative to values in
+            the calling Dataset. Thus by default a 1% deviation of values in da_merge
+            from the calling Dataset is tolerated.
+        error_on_discrepancy (optional), default = True
```
Comment (mikapfl) [inline primap2/_merge.py:261]:
```suggestion
        error_on_discrepancy: (optional), default = True
```

**Review by mikapfl — CHANGES_REQUESTED** [review]
Thanks for finding the problem + fixing it. I'd appreciate a test, since I don't fully understand what is going on and why `how = "all"` doesn't do what I thought it would. With a test case, we could make sure we're not getting regressions later on. But it might be difficult to engineer a nice test case.

Other changes that I requested are just typos I think and an inadvertently included unrelated change.

**Comment by JGuetschow** [comment]
I will add a test. I was also puzzled that "all" didn't do what it supposedly should. Maybe it's a bug. When we have a minimal test we can investigate further.

**Comment by mikapfl** [comment]
I think we can merge the fix also without the minimal test case. Better have the fix without a test than to have an unfixed primap.

**Comment by JGuetschow** [comment]
I think my description is wrong. Only one of the two issues was present, namely keeping the all-nan timeseries. The dropping of singe data points was a mistake by me, I think. Thus I think merging is fine without a test, because the fixed issue is actually small. The larger issue did not exist in the first place, I think.

**Pull request merged**